File: jvm/core/src/main/java/se/idkollen/client/models/MitIdSignRequest.java
package se.idkollen.client.models;

import org.jspecify.annotations.Nullable;

/**
 * Request body for starting a MitID signing session.
 */
public class MitIdSignRequest {
    public String text = "";
    public @Nullable String redirectUrl;
    public @Nullable String refId;

    public MitIdSignRequest setText(String v) {
        text = v;
        return this;
    }

    public MitIdSignRequest setRedirectUrl(@Nullable String v) {
        redirectUrl = v;
        return this;
    }

    public MitIdSignRequest setRefId(@Nullable String v) {
        refId = v;
        return this;
    }
}

File: php/src/Models/MitId/MitIdSignRequest.php
<?php

declare(strict_types=1);

namespace Idkollen\Client\Models\MitId;

readonly class MitIdSignRequest
{
    public function __construct(
        public string $text,
        public ?string $redirectUrl = null,
        public ?string $refId = null,
    ) {}
}
